MovieQuiz/Services/QuestionFactory.swift

```diff
@@ -0,0 +1,113 @@
+
+import Foundation
+
+class QuestionFactory: QuestionFactoryProtocol {
+    
+    private let moviesLoader: MoviesLoading
+    private var movies: [MostPopularMovie] = []
+    weak var delegate: QuestionFactoryDelegate?
+    
+    func loadData() {
+        moviesLoader.loadMovies { result in
```

Comment:
```suggestion
        moviesLoader.loadMovies { [weak self] result in
```

Здесь стоит захватить [weak self] у первого кложура тоже. На самом деле конкретно этот код к утечкам памяти не приводит, однако, если бы это замыкание сохранялось внутри MoviesLoader'а, то контекст захватывался сильно. Так как MoviesLoader этого не делает, все в порядке. Однако, с точки зрения клиентского кода относительно MoviesLoader, мы не знаем сохраняет он замыкание или нет, поэтому лучше захватить контекст слабо.